### PR TITLE
Fix measureFont

### PR DESCRIPTION
**Pull request by jflcarvalho**
ISSUE: #7030  

##### Description of change
Fix measureFont and optimize to use modern browsers functionalities, in some cases the result properties are wrong.

##### Pre-Merge Checklist

- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Review by themoonrat — CHANGES_REQUESTED** [review]
I appreciate the attempt to have a fix of this.

A couple of worries I have tho

1: actualBoundingBoxAscent is marked by MDN as 'Experimental. Expect behavior to change in the future.' - https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics/actualBoundingBoxAscent - I'm unsure if this is something we'd want to put into PixiJS until it's a little more mature.
2: This is going to change how existing text looks. We can't afford to have a minor version of PixiJS effect the positioning, as there is a lot of content out there relying on it's current behaviour. Also, this will change formatting just from browser to browser, but even from browser version to browser version. Right now in PixiJS, all text renders the same across all browsers.

So for me, any fix needs to demonstrate that existing text is not effected, only 'buggy' text fixed, and that this is consistent across all browsers.

Cheers!

**Comment by jflcarvalho** [comment]
Hi,

Yeah I understand, the use of actualBoundingBoxAscent it has just a try to improve the method's performance, and can be deleted.

I tried on some developments that I have on my machine with safari and chrome and it has everything at same place as before, but I can't test on every browser :/

With that I just propose a increase on canvas height and Y placement of the textString to solve this issue.

**Comment by bigtimebuddy** [comment]
The TextMetrics are not that precise. That's why we have overrides like Baseline Multiplier so that users and adjust for their particular
typefaces. I believe that describes your case. We choose these defaults as accommodating most (but not all) circumstances. Changing those values is a breaking change. 

Why not create a new variable for the 2x (2.25x) value and then you should be able to solve your issue without the other changes.

**Comment by jflcarvalho** [comment]
As requested the new changes are applied, I have create a new static variable to defining height multiplier of canvas, with the default value being 2.0x.

File: packages/text/src/TextMetrics.ts [hunk 1]
```diff
@@ -643,15 +644,16 @@ export class TextMetrics
...
-        const canvas = TextMetrics._canvas;
         const context = TextMetrics._context;
 
         context.font = font;
-
         const metricsString = TextMetrics.METRICS_STRING + TextMetrics.BASELINE_SYMBOL;
-        const width = Math.ceil(context.measureText(metricsString).width);
+
+        const textMetrics = context.measureText(metricsString);
+        const canvas = TextMetrics._canvas;
```
Comment (themoonrat) [inline packages/text/src/TextMetrics.ts:653]: slightly pedantic, but if you move
this const canvas = TextMetrics._canvas;
line back to where it was, about the  const context = TextMetrics._context; line, it reduces the diff down to just the changes required, and reads a bit better.

**Review by themoonrat — CHANGES_REQUESTED** [review]
Just a little comment of reducing the diff a bit, but I think this is a good strategy. Keeps current compatibility, whilst allows developers to make the call themselves to measure in a way more appropriate for themselves.
Thanks!

File: packages/text/src/TextMetrics.ts (hunk 1 above)
Comment (jflcarvalho) [reply packages/text/src/TextMetrics.ts:653]: Yah sure!

**Comment by jflcarvalho** [comment]
@themoonrat the requested changes are done!

**Review by themoonrat — APPROVED** [review]

**Review by bigtimebuddy — APPROVED** [review]
Thank you!

**Comment by bigtimebuddy** [comment]
Thank you @jflcarvalho!

**Pull request merged**